### PR TITLE
Enable RBAC without AAD

### PR DESCRIPTION
**Pull request by raphaelquati**
Implementation for #2345

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
hi @raphaelquati 

Thanks for this PR :)

During the investigation into RBAC recently it appeared that whilst you can enable/disable RBAC for AKS at this time it's tied to AzureAD. For this reason we've designed the block such that if/when additional methods are available for RBAC in the future that we can support them as nested items in the `role_based_access_control` block, by making the AzureAD block optional.

As such I believe we should be able to infer the value of the `enable_rbac` field and as such we should be able to not expose it - do you have a reference which shows that it's possible to Enable RBAC without Azure AD?

Thanks!

**Comment by joakimhellum-in** [comment]
To be clear: I completely agree on RBAC being related to Azure AD for user authentication, and we are currently re-deploying all our clusters to enable this. Also because it's in the security [best practices](https://github.com/Azure/k8s-best-practices/blob/master/Security_securing_a_cluster.md#master-endpoint-security) .

However it is possible to only enable RBAC without Azure AD integration at the moment.

1. Using the Azure Portal we can in fact only enable RBAC and not the Azure AD integration:
image

2. Using Azure CLI RBAC is enabled by default (`--enable-rbac`):
image

But you would need to enable Azure AD by setting the parameters (`--aad-client-app-id`, `--aad-server-app-id`, `--aad-server-app-secret` and `--aad-tenant-id`):
image

Reference: https://docs.microsoft.com/en-us/cli/azure/aks?view=azure-cli-latest#az-aks-create

So I think multiple users might have an issue with this update unless they upgrade their clusters to enable Azure AD integration.

**Comment by raphaelquati** [comment]
@tombuildsstuff,

I try to follow the same "block idea" of others blocks (like oms_agent and http_application_routing), with the "enabled" flag, instead of inferring the presence of block.

File: azurerm/resource_arm_kubernetes_cluster_test.go [hunk 1]
```diff
@@ -596,6 +597,7 @@ resource "azurerm_kubernetes_cluster" "test" {
   }
 
   role_based_access_control {
+		enaled=true
```
Comment (timfpark) [inline azurerm/resource_arm_kubernetes_cluster_test.go]: it looks like you have a typo here: enaled -> enabled

**Comment by ams0** [comment]
Please prioritize this together with #2421 , the current implementation forces AAD integration whenever RBAC is needed (which is most of the times), and that's not ideal nor the AKS defaults. Let me know if I can help testing it.

**Comment by tombuildsstuff** [comment]
As discussed above - closing this in favour of #2495

**Pull request closed without merging**